src/NServiceBus.Core/Unicast/Transport/TransportReceiver.cs

```diff
@@ -142,6 +128,8 @@ protected virtual void InnerStop()
             }
 
             receiver.Stop();
+            pipeline.Cooldown();
```

Comment:
I'd say push the task as high up in the call stack as possible. So make pipeline.Cooldown(); return a task as well and synchronize here or even push it up more